transcom/mymove
# B 21374 ub fields int

> [!IMPORTANT]
> This ticket has a LOT of AC. Please do not rush this review.

## [B-21374](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1014669&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

Introduces the optional and OCONUS specific fields of accompanied tour and dependent under/over 12 years of age. Feature flags the frontend portion of the application - the backend is not FF'd as it is the internal API and optional, nullable fields with no intent to ever make not nullable. This means the backend works as intended with or without the FF. The Prime cannot control these fields at this time, this is purely for customer and office user.

These OCONUS specific fields are for UB allowance enabling.

## How to test

Please test with FF on and off

### Setup
1. Think of two different duty locations
2. Once you have two different duty locations, open one up within the database, navigate to its address, and set the address `is_oconus` to true
3. Now you have a CONUS and OCONUS duty location
4. Update `.envrc` and adjust `+export FEATURE_FLAG_UNACCOMPANIED_BAGGAGE=true`

### Customer
1. Navigate to milmovelocal and create a new account
2. Create a new order and fill out all forms until you reach "Tell us about your move orders"
![image](https://github.com/user-attachments/assets/1367f9a4-b66a-4479-b33e-0229a83b5c79)
3. Fill in current and new duty location, ensuring to set one of these locations to the OCONUS version you created in the "Setup" section above.
5. Test adding dependents
6. Remove your OCONUS duty location and check that the OCONUS fields dissappear
7. Try to break it
8. Fully submit your move
9. Review your request and make sure you can edit your orders
10. Try that break that part too
11. Proceed to Office

### Office
1. Once you have submitted the move as a customer, login to Office and preview the submitted move as a SC and TOO
2. The SC should be able to see the information under entitlements and update them
3. The TOO should be able to see the information under entitlements and update them
4. These OCONUS entitlement fields shouldn't be present on CONUS moves, verify that
5. Try to break it


## Recordings
https://github.com/user-attachments/assets/1d8f5474-5b7c-4caa-a20c-77e73c3df358

https://github.com/user-attachments/assets/480c886f-7ee9-4afa-9d0d-a1b9f84ea6f1


## 508 Screenshot
Tooltips do not violate ANDI
![image](https://github.com/user-attachments/assets/2a6fa962-231d-4e45-8380-6af0bb79deef)
